python/lsst/meas/algorithms/sourceSelector.py

```diff
@@ -97,3 +103,301 @@ def _isBad(self, source):
+        """
+        flagField = self.fluxField + "_flag"
+        if flagField in catalog.schema:
+            selected = np.logical_not(catalog[flagField])
+        else:
+            selected = np.ones(len(catalog), dtype=bool)
+
+        flux = catalog[self.fluxField]
+        if self.minimum is not None:
+            selected &= flux > self.minimum
+        if self.maximum is not None:
+            selected &= flux < self.maximum
+        return selected
+
+
+class MagnitudeLimit(pexConfig.Config):
+    """Select sources using a magnitude limit
+
+    Note that this assumes that a zero-point has already been applied and
+    the fluxes are in AB fluxes. It is therefore principally intended for
+    reference catalogs rather than catalogs extracted from science images.
+
+    This object can be used as a `lsst.pex.config.Config` for configuring
+    the limit, and then the `apply` method can be used to test sources in
+    a catalog against the configured limit.
+    """
+    fluxField = pexConfig.Field(dtype=str, default="flux",
+                                doc="Name of the source flux field to use.")
+    minimum = pexConfig.Field(dtype=float, optional=True, doc="Select objects fainter than this magnitude")
+    maximum = pexConfig.Field(dtype=float, optional=True, doc="Select objects brighter than this magnitude")
+
+    def apply(self, catalog):
+        """Apply the magnitude limit to a catalog
+
+        Parameters
+        ----------
+        catalog : `lsst.afw.table.SourceCatalog`
+            Source to which to apply limits.
+
+        Returns
+        -------
+        selected : `numpy.ndarray`
+            Boolean array indicating whether each source is selected.
+        """
+        flagField = self.fluxField + "_flag"
+        if flagField in catalog.schema:
+            selected = np.logical_not(catalog[flagField])
+        else:
+            selected = np.ones(len(catalog), dtype=bool)
+
+        magnitude = lsst.afw.image.abMagFromFlux(catalog[self.fluxField])
+        if self.minimum is not None:
+            selected &= magnitude > self.minimum
+        if self.maximum is not None:
+            selected &= magnitude < self.maximum
+        return selected
+
+
+class RequireFlags(pexConfig.Config):
+    """Select sources using flags
+
+    This object can be used as a `lsst.pex.config.Config` for configuring
+    the limit, and then the `apply` method can be used to test sources in
+    a catalog against the configured limit.
+    """
+    good = pexConfig.ListField(dtype=str, default=[],
+                               doc="List of source flag fields that must be set for a source to be used.")
+    bad = pexConfig.ListField(dtype=str, default=[],
+                              doc="List of source flag fields that must NOT be set for a source to be used.")
+
+    def apply(self, catalog):
+        """Apply the flag requirements to a catalog
+
+        Returns whether the source is selected.
```

Comment:
"Return a boolean array specifying whether each source is selected." (and below)

Reply:
That's what the "Returns" section already says, so I think that's redundant.